src/examples/algorithms/Anomaly.cpp

```diff
@@ -0,0 +1,199 @@
+#include <iostream>     
+#include <fstream>      
+#include <vector>
+#include <string>
+#include <algorithm>    
+#include <iterator>     
+#include <boost/tokenizer.hpp>
+
+#include "nupic/encoders/ScalarEncoder.hpp"
+#include "nupic/algorithms/SpatialPooler.hpp"
+#include "nupic/algorithms/Cells4.hpp"
+#include "nupic/algorithms/Anomaly.hpp"
+#include "nupic/os/Timer.hpp"
+
+
+// Prints a vector as long as it's type has operator << defined.
+template<typename T>
+void print_vector(const std::vector<T>& v, char sep=0)
+{
+    for (auto& elem: v) {
+        std::cout << elem;
+        if (sep != 0) {
+            std::cout << sep;
+        }
+    }
+    std::cout << std::endl;
+}
+
+
+// Converts (casts) all elements of a vector from T1 to T2 (returns a new vector)
+template<typename T1, typename T2>
+std::vector<T2> vectorTypeCast(const std::vector<T1>& orig)
```

Comment:
1) should the 2 above methods be "promoted" to some `algorithms/Utils` helper class, is in python? 
2) converting between `Real and UInt` really sucks, as seen in this example or in https://github.com/numenta/nupic.core/blob/master/src/examples/algorithms/HelloSP_TP.cpp#L71-L78 
Why is that?